core/src/main/java/org/apache/james/core/Username.java

```diff
@@ -33,6 +34,7 @@
 
 public class Username {
     public static final int MAXIMUM_MAIL_ADDRESS_LENGTH = 255;
+    private static String ILLEGAL_USERNAME_CHARACTERS = "\"(),:; <>@[\\]";
```

Comment:
Are we sure of that ?
By this source the quotes are acceptable and the other characters are acceptable if quoted.
https://en.wikipedia.org/wiki/Email_address#Local-part

Reply:
Furthermore this change is conflicting with some tests in `MailAdressTest`  using this list of valid addresses : 
```
    private static final String[] GOOD_ADDRESSES = {
            GOOD_ADDRESS,
            GOOD_QUOTED_LOCAL_PART,
            "server-dev@james-apache.org",
            "server-dev@[127.0.0.1]",
            "server-dev@#123",
            "server-dev@#123.apache.org",
            "server.dev@james.apache.org",
            "\\.server-dev@james.apache.org",
            "server-dev\\.@james.apache.org",
    };
```

Reply:
I think we can control the creation of the user in our domain but we should accept mails from other domains with more complicated local parts.

Reply:
Using Username to represent external entities seems wrong.

Where does this happens?

Reply:
Btw thats why Username != MailAddress.

I think we are good...

Reply:
The problem is that it's difficult to be sure : 
I think i have seen at least a potential issue in  : 
`SpamAssassin.service` where we check for each recipient and in the `querySpamAssassin` we transform the recipient mail address to a username.

Reply:
What about the existing malformed usernames ? Will it blow everything to add this precondition without a migration plan ?

Reply:
> in the querySpamAssassin we transform the recipient mail address to a username

All the wrong usages should be fixed.

> What about the existing malformed usernames ? Will it blow everything to add this precondition without a migration plan ?

Good point.

Reply:
And finally, we at linagora doesn't see the value in quoted local part (with good reasons ok) but we are not sure that some james user in the wild doesn't use this heavily , maybe we should at least ask our userbase about this change ?

Reply:
I know, according to the RFC, it's supposed to be allowed if those characters are escaped with a backslash or if the local part if between quotes... we did debate about that with Benoit.

But honestly so far in the actual code, regarding the Username, some of those characters are being accepted without the quotes or backslash, some are not, and there is no quote or backslash implementation regarding those special characters.

I looked a bit and implementing this seems IMO quite a bit of a headache for not much value added (so what you can have a fancy character in your mail address? I hardly understand why the RFC allows such a thing in the first place)... You have other email providers also that decided to forbid those for probably the same reasons. 

If you think that it's ok to keep enforcing this rule for the simplicity of the code but that it's better to ask the community first, I'm ok.
If you think it's better to not touch anything, to not break possible usernames for some james users in the wild, ok too, even if it feels like we don't have a position regarding this matter.
If you think we should follow the RFC, well please explain your solution, cause it doesn't look simple enough to me for too little value :)